### DISCUSSION TITLE
ActivityPub support / Join Fediverse

### DISCUSSION THREAD

**Discussion by rugk** [forem/forem#15316] [Feature Requests]
### User Story / Description

As a user of Dev.io/Mastodon/Peertube/…, I want a federated web, where I can follow anyone on any instance of a "social network" (or similar), because it is what the web is like/should be and is a usability improvement.

For this to work, you should try to add ActivityPub support, as this is the new standardized federation protocol. See:
* https://activitypub.rocks/
* https://en.wikipedia.org/wiki/ActivityPub
* https://blog.joinmastodon.org/2018/06/why-activitypub-is-the-future/

### Definition of Done

* ActivityPub support implemented
* Sharing stuff to other networks work.
* You can follow other user's on other instances.
* Maybe more(?).

**Comment by michaelgv** [comment]
It would be nice, as the posts could be synced over to fediverse, but it begs the question, if someone interacts on the fediverse, should Dev's platform automatically insert that reaction, or should it leave that door closed? If we go with the former, it's a tad bit of work to get it properly implemented, and effectively it's "shadow" accounts; syncing could be become a very bad nightmare quickly.

**Comment by ortegacmanuel** [comment]
I would love to work on this. I had previous experience implementing ActivityPub on other projects. But first I would like to know what project owners think about this feature cc @benhalpern If they like the idea I could start working on this next week :)

**Comment by benhalpern** [comment]
I'm in favor of this sort of thing. In the past I've looked into this but didn't come away with specific product inspiration. @michaelgv's questions are definitely where my head goes. We'd have to do this the right way.

So I'm very excited by this and I think it could be really fun to work out as a community, but I think there are a lot of conceptual details to hammer out.

So let's commit to this as an ongoing discussion.

**Comment by ortegacmanuel** [comment]
@benhalpern thanks for your reply. Recently [Write.as added support to ActivityPub](https://discuss.write.as/t/activitypub-support/64). They added a basic implementation as described in the next list 

- Finding Write.as blogs by name from within Mastodon, e.g. @blog@write.as for write.as/blog/ or blog.writeas.com 4
- Following Write.as blogs
- Getting blog posts in your home stream / on your instance’s federated timeline
- Favoriting posts (nothing happens on our end – we consider it a private bookmark)
- Boosting posts

I think that it is a good first step forward on adding support to ActivityPub, just enabling broadcasting of Dev.to content across Federated Web using ActivityPub.

I'm now exploring the Deb.to codebase in order to see how this could be easily added and happy to be part of that ongoing discussion.

**Comment by stale** [comment]
Thanks for contributing to this issue. As it has been 90 days since the last activity, we are automatically closing the issue in 7 days. This is often because the request was already solved in some way and it just wasn't updated or it's no longer applicable. If this issue still requires attention, please respond with a comment. Happy Coding!

**Comment by rugk** [comment]
No bot, keep it open. :smiley:

**Comment by zcdunn** [comment]
[Prismo](https://prismo.news) is a link-sharing application (think reddit, hacker news, lobsters) that federates over ActivityPub. It's also written in Ruby and the developer is working on extracting the AP support into a library. Anybody interested in this issue could jump over to [his repo](https://gitlab.com/prismosuite/activitypub-rails/) and help out there to provide an easy path for dev.to to integrate AP

**Comment by stale** [comment]
Thanks for contributing to this issue. As it has been 90 days since the last activity, we are automatically closing the issue in 7 days. This is often because the request was already solved in some way and it just wasn't updated or it's no longer applicable. If this issue still requires attention, please respond with a comment. Happy Coding!

**Comment by maestromac** [comment]
not stale

**Comment by aschrijver** [comment]
**FYI**: I just [described a vision](https://meta.discourse.org/t/132624/40?u=aschrijver) for Discourse as-if it were built from the ground up with federation support, called **"Community has no boundary"** and which turn Discourse from a stand-alone product into Discourse-as-a-Fabric. It may be inspiring to **forem** to consider these possibilities too. Also adopting a compatible federation design would allow `forem <----> discourse` community integration.

**Edit**: I have added forum to the ~~ActivityPub watchlist~~ [delightful-fediverse-apps](https://codeberg.org/fediverse/delightful-fediverse-apps) and we'll monitor this issue for progress.

**Comment by resynth1943** [comment]
Thanks, A! :heart: I put out a [post](https://mastodon.tedomum.net/@resynth1943/105467385328376766) on my Mastodon, so hopefully this gets picked up soon. Would love for dev.to to interact with an interconnected network of over 4 million people. This would do *wonders* for the JS community on Mastodon, too :heart: 

Will be keeping my Eagle Eye :tm: on this issue.

**Comment by aschrijver** [comment]
FYI there's likely to be movement on a Discourse ActivityPub impl. in 2022. More info: https://socialhub.activitypub.rocks/t/discourse-will-be-joining-the-fediverse/2071

**Comment by jcklpe** [comment]
This sounds like it would be really cool.

**Comment by aschrijver** [comment]
FYI [fediverse.party](https://fediverse.party) has migrated to Codeberg, and here you'll also find the Delightful lists that provide input to the side and contain the latest known state of ongoing projects: https://codeberg.org/fediverse

**Comment by phlash** [comment]
Adding my support to this suggestion, partly as I have just joined the fediverse and this would directly improve the UX for me and mostly to support the idea that it's a reasonable way to surface a feed of tagged/filterable articles to many, many more people (after the Twitter exodus of the last few weeks!) in a socially acceptable manner.

**Comment by ajsirota** [comment]
With the rise in fediverse usage from the mass migration off the Website Formerly Known As Twitter, is there any plan to add ActivityPub support to Forem? It's making some members of a community I'm in hesitant to adopt it.

**Comment by aschrijver** [comment]
I can add to that that Discourse recently released an ActivityPub plugin, and both NodeBB and Flarum are working on adding AP support. Forum have somewhat different needs and perspectives and their integration is sometimes referred to as forming the "threadiverse" (though I find this distinction to be unnecessary). There's also a first Fediverse Enhancement Proposal (FEP) relating to inter-forum communication: https://socialhub.activitypub.rocks/t/fep-d36d-sharing-content-across-federated-forums/3366

**Comment by kamazeuci** [comment]
Are there any news about ActivityPub implementation in Forem? are the developers interested? is it on the roadmap?

**Comment by rugk** [comment]
In other news, NTW, the FLOSS forum software [**ImgBB** has joined the Fediverse](https://techcrunch.com/2025/01/21/forum-software-nodebb-joins-the-fediverse/) aka implemented ActivityPub.
